vars/buildDockerAndPublishImage.groovy

```diff
@@ -73,11 +91,35 @@ def call(String imageName, Map config=[:]) {
         } // steps
       } //stage
 
+      stage("Semantic Release") {
+        when {
+          allOf {
+            expression { env.SEMANTIC_RELEASE == 'true' }
+            branch dockerConfig.mainBranch  
+          }
+        }
+        steps {
+          echo "Configuring credential.helper"
+          sh 'git config --local credential.helper "!f() { echo username=\\$GIT_USERNAME; echo password=\\$GIT_PASSWORD; }; f"'
+
+          withCredentials([usernamePassword(credentialsId: "${dockerConfig.gitCredentials}", passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
```

Comment:
Is there a reason to not use the environment form?

e.g.

```
stage("Semantic Release") {
  when {
    allOf {
      expression { env.SEMANTIC_RELEASE == 'true' }
      branch dockerConfig.mainBranch  
    }
  }
  environment {
    GITHUB_CREDENTIALS = credential("${dockerConfig.gitCredentials}")
  }
  steps {
    echo "Configuring credential.helper"
    sh 'git config --local credential.helper "!f() { echo username=\\$GITHUB_CREDENTIALS_USR; echo password=\\$GITHUB_CREDENTIALS_PSW; }; f"'
    echo "Tagging New Version: $nextVersion"
    sh "git tag $nextVersion"

    echo "Pushing Tag"
    sh "git push origin --tags"
  }
}
```

Reply:
Yeah, only cause AFAIK the environment form is not implemented in JenkinsPipelineUnit.